task_test.go

```diff
@@ -24,12 +25,20 @@ import (
 
 func TestAllTasks(t *testing.T) {
 	client := NewFakeMarathonEndpoint(t)
-	tasks, err := client.AllTasks()
+	tasks, err := client.AllTasks(nil)
 	assert.Nil(t, err)
 	assert.NotNil(t, tasks)
 	assert.Equal(t, len(tasks.Tasks), 2)
 }
 
+func TestAllStagingTasks(t *testing.T) {
+	client := NewFakeMarathonEndpoint(t)
+	tasks, err := client.AllTasks(url.Values{"status": []string{"staging"}})
```

Comment:
@eicca sorry for the delay ... I guess my reasoning behind I didn't fancy chasing the API. Side stepping just AllTasks() for moment has one; starts off with one variable, then another comes along, so you add it and break it all other builds (yes, we have godeps for that, but), then perhaps the second option is supported in one version but not the early release's etc etc .. and then times by x methods, i've a feeling it could get messy. Wrapping them in structs does mitigate most of that, it's simply a question if you want those pretty (e.g marathon.TaskStaging) or opened up by using a url.Values and letting the user decide ...